edx/edx-enterprise
# [ENT-628] [ENT-629] [ENT-630] [ENT-631] [ENT-634] Display Program Landing Page Details

**Description:** This PR adds the remaining program details on the program enrollment landing page that we need. This includes:

- "What is `PLATFORM_NAME`?" on the sidebar, with a configurable `PLATFORM_DESCRIPTION`.
- Program type description on the sidebar, with the program type logo.
- Configurable `ENTERPRISE_TAGLINE` on the page header for our landing pages plus our data sharing consent page.
- A "What you'll learn" section, that is expandable for when there are multiple expected learning outcomes.
- A "Real Career Impact" section, that is expandable for when there are multiple corporate endorsements.
- Courses on the course list in the program summary box have pacing information below them.
- Courses on the course list in the program summary box can be clicked to bring up our course modal (the same one from the course enrollment landing page).
- Price data, as well as a discount provider, is written in the "Price" row of the program summary box.

**JIRA:** [ENT-628](https://openedx.atlassian.net/browse/ENT-628), [ENT-629](https://openedx.atlassian.net/browse/ENT-629), [ENT-630](https://openedx.atlassian.net/browse/ENT-630), [ENT-631](https://openedx.atlassian.net/browse/ENT-631), & [ENT-634](https://openedx.atlassian.net/browse/ENT-634).

**Merge deadline:** End of sprint.

**Testing instructions:**

### Setup

1. Make authoring organizations on course-discovery.
1. Make multiple corporate endorsements on course-discovery.
1. Make courses (audit and verified track) and course runs, with filled out information for most fields, and use any authoring organization if applicable. Make them appear on course-discovery.
1. Make a program type -- one of `Professional Certificate`, `MicroMasters Certificate`, or `XSeries Certificate`.
1. Make a program containing anything you created above, if applicable (and note the UUID). Make it one-click purchasable, and add expected learning outcomes.
1. Make a program API config (actually, it's just a "program config", but we aren't changing the name yet on the model) at LMS path `/admin/programs/programsapiconfig/`. Make sure it's enabled.
1. Make SKU data associated with the course(s) you made above on ecommerce.
1. Make an enterprise customer. Note the UUID.

### Testing Program Details

1. Login to the LMS as any user who is not enrolled in the courses & program you created.
1. Navigate to `/enterprise/{enterprise_uuid}/program/{program_uuid}/enroll/`. Replace `{enterprise_uuid}` and `{program_uuid}` with the enterprise's UUID and the program's UUID that you noted above, respectively.
1. Make sure you see the following information (depending on your configuration):

#### Tagline

![selection_005](https://user-images.githubusercontent.com/10018065/30351666-f202bff8-9835-11e7-9fbb-72c78b57c7d5.png)

#### Sidebar

![selection_007](https://user-images.githubusercontent.com/10018065/30351668-f63defca-9835-11e7-93fa-4460456c364c.png)

#### What you'll learn (unexpanded)

![selection_008](https://user-images.githubusercontent.com/10018065/30351678-0545e86a-9836-11e7-830e-8af5e4773b29.png)

#### What you'll learn (expanded)

![selection_009](https://user-images.githubusercontent.com/10018065/30351684-0ac59bf0-9836-11e7-9519-998e38f536eb.png)

#### Real Career Impact (unexpanded)

![selection_010](https://user-images.githubusercontent.com/10018065/30351692-13c98bb2-9836-11e7-9026-a070f00d6ff3.png)

#### Real Career Impact (expanded)

![selection_011](https://user-images.githubusercontent.com/10018065/30351700-1a7f5144-9836-11e7-9a9e-b04089546b81.png)

#### Program Summary Box (course pacing, clickable course titles, and discount provider)

![selection_012](https://user-images.githubusercontent.com/10018065/30351703-29641b2c-9836-11e7-91d6-58f05de8dcb8.png)

#### Course Modal (initial open, course 1)

![selection_013](https://user-images.githubusercontent.com/10018065/30351710-334d6152-9836-11e7-907e-5f77b1bae097.png)

#### Course Modal (scrolled down completely, course 1)

![selection_014](https://user-images.githubusercontent.com/10018065/30351717-3c044752-9836-11e7-8c9d-156705d4852a.png)

#### Course Modal (initial open, any other course, i.e. course 2)

![selection_015](https://user-images.githubusercontent.com/10018065/30351729-46516b0e-9836-11e7-8271-b86c02f7e511.png)

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
